mm/mm_heap/mm_memalign.c

```diff
@@ -106,13 +110,13 @@ FAR void *mm_memalign(FAR struct mm_heap_s *heap, size_t alignment,
 
   /* Then malloc that size */
 
-  rawchunk = (size_t)mm_malloc(heap, allocsize);
-  if (rawchunk == 0)
+  rawchunk_ptr = mm_malloc(heap, allocsize);
```

Comment:
I still think that introduction of two local variables `ptr` and `rawchunk_ptr` is redundant and we can reuse `(FAR void *)` cast of existing `uintptr_t`, but that is minor and more related to code readability than to functionality.